egs/gigaspeech/s5/run.sh

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env bash
+# Copyright 2021 Xiaomi Corporation (Author: Yongqing Wang)
+# Apache 2.0
+
+. ./cmd.sh
+. ./path.sh
+
+stage=0
+
+gigaspeech_root=~/GigaSpeech_data/
+train_sets=train
+test_sets="dev test"
+lm_dir=data/local/lm
+dict_dir=data/local/dict
+
+. ./cmd.sh
+. ./path.sh
+. parse_options.sh
+
+# you might not want to do this for interactive shells.
+set -e
+
+for f in $lm_dir $dict_dir; do
+  [ ! -d $f ] && mkdir -p $f
+done
+g2p_model_dir=$gigaspeech_root/dict/g2p
+
+if [ $stage -le 0 ]; then
+  echo -e "======Download GigaSpeech START|current time : `date +%Y-%m-%d-%T`======"
+  # Download and analyze GigaSpeech datasets
+  # cd GigaSpeech to run utils/gigaspeech_download.sh to download data
+  git clone https://github.com/SpeechColab/GigaSpeech.git
+  pushd GigaSpeech
+  utils/gigaspeech_download.sh $gigaspeech_root
+  popd
+  echo -e "======Download GigaSpeech END|current time : `date +%Y-%m-%d-%T`======"
+fi
+
+# This stage maybe lead to errors,
+# because stage1 in gigaspeech_data_prep.sh should be run in path of gigaspeech
+# stage 3 in gigaspeech_data_prep.sh should be run here
+if [ $stage -le 1 ]; then
+  echo -e "======Prepare Data START|current time : `date +%Y-%m-%d-%T`======"
+  # Prepare GigaSpeech data
+  GigaSpeech/toolkits/kaldi/gigaspeech_data_prep.sh $gigaspeech_root data true
+  echo -e "======Prepare Data END|current time : `date +%Y-%m-%d-%T`======"
+fi
+
+if [ $stage -le 2 ]; then
+  echo -e "======Prepare Dict START|current time : `date +%Y-%m-%d-%T`======"
+  # Prepare dict 
+  # If the lexicon is downloaded from GigaSpeech, you can skip the G2P steps with "--stage 4" option
+  # Otherwise, you should start from "--stage 0" option to use G2P
+  [ ! -f $g2p_model_dir/g2p.model.4 ] && echo "$0: Cannot find $g2p_model_dir/g2p.model.4" && exit 1
+  if [ -f $dict_dict/lexicon_raw_nosil.txt ]; then
+    # If GigaSpeech has been dowloaded, skip G2P steps with "--stage 4"
+    local/prepare_dict.sh --stage 4 $g2p_model_dir/g2p.model.4 data/local/dict
+  else
+    local/prepare_dict.sh --stage 0 --cmd "$train_cmd" --nj 300 --train-set train --test-sets "$test_sets" $g2p_model_dir/g2p.model.4 data/local/dict
```

Comment:
we need to `mkdir -p $dict_dir/cmudict; cp $gigaspeech_root/dict/cmudict.0.7a $dict_dir/cmudict` before `local/prepare_dict.sh`,